internal/controller/deployment_to_pdb_controller.go

```diff
@@ -163,16 +165,15 @@ func (r *DeploymentToPDBReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&v1.Deployment{}).
 		WithEventFilter(predicate.Funcs{
 			UpdateFunc: func(e event.UpdateEvent) bool {
-				//logger.Info("Update event detected, no action will be taken")
-				//ToDo: distinguish scales from our EvictionAutoScaler from scales from other owners and keep minAvailable up near replicas.
-				// Like if I start a deployment at 3 but then later say this is popular let me bump it to 5 should our pdb change.
 				if oldDeployment, ok := e.ObjectOld.(*v1.Deployment); ok {
 					newDeployment := e.ObjectNew.(*v1.Deployment)
-					logger.Info("Update event detected, num of replicas changed", "newReplicas", newDeployment.Spec.Replicas)
-					return oldDeployment.Spec.Replicas != newDeployment.Spec.Replicas
+					if lo.FromPtr(oldDeployment.Spec.Replicas) != lo.FromPtr(newDeployment.Spec.Replicas) {
```

Comment:
this is the main bug fix.

Reply:
unittest?

Reply:
E2e logs demonstrate tehre are now way less firing of this controller. But also AI genning a unittest for this